core/src/main/java/io/questdb/griffin/SqlOptimiser.java

```diff
@@ -2753,6 +2808,43 @@ private void resolveJoinColumns(QueryModel model) throws SqlException {
         }
     }
 
+    //Rewrite:
+    // sum(x*10) into sum(x) * 10, etc.
+    // sum(x+10) into sum(x) + count(x)*10
+    // sum(x-10) into sum(x) - count(x)*10
+    private ExpressionNode rewriteAggregate(ExpressionNode agg, QueryModel model) {
+        if (agg == null) {
+            return null;
+        }
+
+        ExpressionNode op = agg.rhs;
+
+        if (agg.type == FUNCTION &&
+                functionParser.getFunctionFactoryCache().isGroupBy(agg.token) &&
+                Chars.equalsIgnoreCase("sum", agg.token) &&
```

Comment:
Why do we optimize sum only? avg, min, max expressions may be also re-written to keep a single column as the argument.

Reply:
That's to check validity/usefulness of such transformations .

Reply:
> That's to check validity/usefulness of such transformations .

Do we plan to rerun ClickBench or validate the usefulness in any other way then?

Reply:
Yes, plan is to re-run  it after release .